### PR TITLE
Add deprecated annotations

### PR DESCRIPTION
**Pull request by aneeshafedo**
Resolves https://github.com/ballerina-platform/ballerina-openapi/issues/471

## Goals
Scenarion 1 : Deprecate operation

```ballerina
    # List all pets
    #
    # + 'limit - How many items to return at one time (max 100)
    # + return - An paged array of pets
    #
    # # Deprecated
    # This methods has deprecated since the Pet has deprecated
    @deprecated
    remote isolated function listPets(int? 'limit = ()) returns Pets|error {
        string  path = string `/pets`;
        map<anydata> queryParam = {"limit": 'limit};
        path = path + check getPathForQueryParam(queryParam);
        Pets response = check self.clientEp-> get(path, targetType = Pets);
        return response;
    }
```

Scenario 2 : Deprecated parameter

```ballerina
    # Info for a specific pet
    #
    # + petId - The id of the pet to retrieve
    # + 'limit - The id of the pet to retrieve
    # # Deprecated parameters
    # + 'limit - Limit is deprecated from v2
    # + return - Expected response to a valid request
    remote isolated function showPetById(string petId, @deprecated string 'limit) returns Pets|error {
        string  path = string `/pets/${petId}`;
        map<anydata> queryParam = {"limit": 'limit};
        path = path + check getPathForQueryParam(queryParam);
        Pets response = check self.clientEp-> get(path, targetType = Pets);
        return response;
    }
```

Scenario 3 : Deprecated schema

```ballerina
# Pet Object
# 
# # Deprecated
# Pet object is deprecated from version 2
@deprecated
public type Pet record {
    int id;
    string name;
    string tag?;
    string 'type?;
};
```

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: openapi-cli/src/test/resources/generators/client/ballerina/display_and_deprecated.bal [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            }
+            param[param.length()] = "=";
+                                                                                                              if  value  is  string {
+                                                                                                              string updateV =  check url:encode(value, "UTF-8");
+                                                                                                              param[param.length()] = updateV;
+                                                                                                              } else {
+                                                                                                              param[param.length()] = value.toString();
+                                                                                                              }
+                                                                                                              param[param.length()] = "&";
+  }
```
Comment (LakshanSS) [inline openapi-cli/src/test/resources/generators/client/ballerina/display_and_deprecated.bal:84]: Formatting issue.
Is it done on purpose or a mistake?

Comment (aneeshafedo) [reply openapi-cli/src/test/resources/generators/client/ballerina/display_and_deprecated.bal:84]: Its a mistake. Thank you

File: openapi-cli/src/test/resources/generators/client/ballerina/display_and_deprecated.bal [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    # + 'limit - The id of the pet to retrieve
+    # # Deprecated parameters
+    # + 'limit - Limit is deprecated from v2
+    # + return - Expected response to a valid request
+    #
+    # # Deprecated
+    @deprecated
+  remote isolated function showPetById(string petId, @display {label: "Result limit"} @deprecated string 'limit) returns Pets|error {
+                                                                        string  path = string `/pets/${petId}`;
+                                                                        map<anydata> queryParam = {"limit": 'limit};
```
Comment (LakshanSS) [inline openapi-cli/src/test/resources/generators/client/ballerina/display_and_deprecated.bal:53]: Formatting issue

File: openapi-cli/src/test/resources/generators/client/ballerina/deprecated_functions.bal [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    public isolated function init(http:ClientConfiguration clientConfig =  {}, string serviceUrl = "https://petstore.swagger.io:443/v2") returns error? {
+        http:Client httpEp = check new (serviceUrl, clientConfig);
+        self.clientEp = httpEp;
+    }
+    # List all pets
+    #
+    # + 'limit - How many items to return at one time (max 100)
+    # + return - An paged array of pets
+    #
+    # # Deprecated
```
Comment (lnash94) [inline openapi-cli/src/test/resources/generators/client/ballerina/deprecated_functions.bal:20]: why does it have double "#" there, is there any reason for it?

Comment (aneeshafedo) [reply openapi-cli/src/test/resources/generators/client/ballerina/deprecated_functions.bal:20]: It is how Ballerina accepts comments related to `Deprecated` annotations. Removing one `hash` token will be resulted in syntax errors. Please see ballerina by example [here](https://ballerina.io/1.2/learn/by-example/deprecation).

File: openapi-cli/src/test/resources/generators/client/file_provider/swagger/display_annotation.yaml [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            $ref: '#/components/schemas/Tag'
+        status:
+          type: string
+          description: pet status in the store
+          enum:
+            - available
+            - pending
+            - sold
+      xml:
+        name: Pet
```
Comment (lnash94) [inline openapi-cli/src/test/resources/generators/client/file_provider/swagger/display_annotation.yaml:122]: Please add a new line :)

**Comment by lnash94** [comment]
if possible shall we check why coverage will decrease by 2.75% and add some tests for this?

**Review by indikasampath2000 — APPROVED** [review]

**Review by lnash94 — APPROVED** [review]

**Pull request merged**